cms/models.py

```diff
@@ -2635,3 +2637,104 @@ def get_context(self, request, *args, **kwargs):
                 "HUBSPOT_ENTERPRISE_PAGE_FORM_ID"
             ),
         }
+
+
+class ExternalCoursePagePlatform(models.Model):
+    """
+    Abstract model for platform specific pages
+    """
+
+    platform = models.ForeignKey(
+        Platform, on_delete=models.SET_NULL, null=True, blank=True
+    )
+
+    class Meta:
+        abstract = True
+
+
+class CommonExternalCoursePage(CanCreatePageMixin, DisableSitemapURLMixin, Page):
+    """
+    A placeholder class to group CommonExternalCoursePages as children.
+    This class logically acts as no more than a "folder" to organize
+    pages and add parent slug segment to the page url.
+    """
+
+    slug = COMMON_EXTERNAL_COURSE_SLUG
+
+    parent_page_types = ["CourseIndexPage"]
+
+    def serve(self, request, *args, **kwargs):  # noqa: ARG002
+        """
+        For index pages we raise a 404 because these pages do not have a template
+        of their own and we do not expect a page to available at their slug.
+        """
+        raise Http404
+
+
+class CommonExternalCoursePageMixin:
```

Comment:
Maybe? I think we can have better name.
```suggestion
class CommonChildPageMixin:
```